plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -226,6 +228,10 @@ else if (config.getPinGlueClientToCurrentRegion()) {
             asyncGlueClientBuilder.setRegion(getCurrentRegionFromEC2Metadata().getName());
         }
 
+        if (config.getAwsCredentialsProviderConf().isPresent()) {
+            checkArgument(config.getAwsCredentialsProvider().isPresent(), "AwsCredentialsProvider class must be set when AwsCredentialsProviderConf is set");
```

Comment:
nit : Can we move these validations to `GlueHiveMetastoreConfig` - we could have an public method with `@AssertTrue` annotation where we could check if configurationFile is provided then we should specify `AwsCredentialsProvider` class